utilities/getRelativeDateString.ts

```diff
@@ -1,50 +1,52 @@
 export function getRelativeDateString(date: Date): string {
   const now = new Date();
 
-  const yearDiff = now.getFullYear() - date.getFullYear();
+  const diff = (now.getTime() - date.getTime()) / 1000;
+
+  const yearDiff = Math.floor(diff / (60 * 60 * 24 * 365));
   if (yearDiff) return `${yearDiff} ${yearDiff === 1 ? "year" : "years"}`;
 
-  const monthDiff = now.getMonth() - date.getMonth();
+  const monthDiff = Math.floor(diff / (60 * 60 * 24 * 30));
   if (monthDiff) return `${monthDiff} ${monthDiff === 1 ? "month" : "months"}`;
 
-  const dayDiff = now.getDate() - date.getDate();
-  const weekDiff = Math.floor(dayDiff / 7);
+  const weekDiff = Math.floor(diff / (60 * 60 * 24 * 7));
   if (weekDiff) return `${weekDiff} ${weekDiff === 1 ? "week" : "weeks"}`;
 
+  const dayDiff = Math.floor(diff / (60 * 60 * 24));
   if (dayDiff) return `${dayDiff} ${dayDiff === 1 ? "day" : "days"}`;
 
-  const hourDiff = now.getHours() - date.getHours();
+  const hourDiff = Math.floor(diff / (60 * 60));
   if (hourDiff) return `${hourDiff} ${hourDiff === 1 ? "hour" : "hours"}`;
 
-  const minuteDiff = now.getMinutes() - date.getMinutes();
+  const minuteDiff = Math.floor(diff / 60);
   if (minuteDiff)
     return `${minuteDiff} ${minuteDiff === 1 ? "minute" : "minutes"}`;
 
-  const secondDiff = now.getSeconds() - date.getSeconds();
-  return `${secondDiff} ${secondDiff === 1 ? "second" : "seconds"}`;
+  return `${diff} ${diff === 1 ? "second" : "seconds"}`;
 }
 
 export function getRelativeShortDateString(date: Date): string {
   const now = new Date();
 
-  const yearDiff = now.getFullYear() - date.getFullYear();
+  const diff = (now.getTime() - date.getTime()) / 1000;
+
+  const yearDiff = Math.floor(diff / (60 * 60 * 24 * 365));
   if (yearDiff) return `${yearDiff} y`;
 
-  const monthDiff = now.getMonth() - date.getMonth();
+  const monthDiff = Math.floor(diff / (60 * 60 * 24 * 28));
   if (monthDiff) return `${monthDiff} m`;
 
-  const dayDiff = now.getDate() - date.getDate();
-  const weekDiff = Math.floor(dayDiff / 7);
+  const weekDiff = Math.floor(diff / (60 * 60 * 24 * 7));
   if (weekDiff) return `${weekDiff} w`;
 
+  const dayDiff = Math.floor(diff / (60 * 60 * 24));
   if (dayDiff) return `${dayDiff} d`;
 
-  const hourDiff = now.getHours() - date.getHours();
+  const hourDiff = Math.floor(diff / (60 * 60));
   if (hourDiff) return `${hourDiff} h`;
 
-  const minuteDiff = now.getMinutes() - date.getMinutes();
+  const minuteDiff = Math.floor(diff / 60);
   if (minuteDiff) return `${minuteDiff} ${minuteDiff === 1 ? "min" : "mins"}`;
 
-  const secondDiff = now.getSeconds() - date.getSeconds();
-  return `${secondDiff} ${secondDiff === 1 ? "sec" : "secs"}`;
+  return `${diff} ${diff === 1 ? "sec" : "secs"}`;
```

Comment:
What's the purpose of these changes? 🤔 Seems not related to the change using date mock

Reply:
That is true. It isn't related to using date mock. I'll delete it and create another issue.

Reply:
Nice, thanks!